src/OrleansSQLUtils/Storage/DbConnectionFactory.cs

```diff
@@ -13,10 +14,21 @@ internal static class DbConnectionFactory
 
         private static CachedFactory GetFactory(string invariantName)
         {
+            if (String.IsNullOrWhiteSpace(invariantName))
+            {
+                throw new ArgumentNullException(nameof(invariantName));
+            }
+
             // Seeks for database provider factory classes from GAC or as indicated by
             // the configuration file, see at <see href="https://msdn.microsoft.com/en-us/library/dd0w4a2z%28v=vs.110%29.aspx">Obtaining a DbProviderFactory</see>.       
 
             DataRow factoryData = DbProviderFactories.GetFactoryClasses().Rows.Find(invariantName);
+
+            if (factoryData == null)
+            {
+                throw new InvalidOperationException($"Can't find database provider factory with '{invariantName}' invariant name. Please check the configuration file to see if the provider is configured correctly or the binaries/assemblies are deployed with the application.");
```

Comment:
Maybe change the part of "binaries/assemblies" to "ADO.NET libraries" as it is more specific about the exact issue.